File: frontend/src/app/interfaces/menu.ts
export interface Menu {
  id: number;
  name: string;
  category: string;
  ingredients: string;
  price: number;
  image: string;
  restaurantId: number;
  additives?: Additives[];
  chooseAdditives?: Additives[];
  quantity?: number;
}


export interface Additives {
  id: number;
  name: string;
  value: string;
  price?: number;
}

File: old/backend/src/main/java/pl/urban/backend/model/Menu.java
package pl.urban.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.Set;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="menus")
public class Menu {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String category;

    private String ingredients;

    @Column(nullable = false)
    private double price;

    private String image;

    @ManyToMany(fetch = FetchType.EAGER, mappedBy = "menus")
    private List<Additives> additives;

    @ManyToMany(mappedBy = "menu")
    private Set<Restaurant> restaurant;

    @OneToMany(mappedBy = "menu", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderMenu> orderMenus;

}
